xds/internal/client/v2client_lds_test.go

```diff
@@ -62,6 +68,29 @@ func (s) TestLDSGetRouteConfig(t *testing.T) {
 			wantRoute: "",
 			wantErr:   true,
 		},
+		{
+			name: "rds.ConfigSource-in-apiListener-is-not-ADS",
+			lis: &xdspb.Listener{
+				Name: goodLDSTarget1,
+				ApiListener: &listenerpb.ApiListener{
+					ApiListener: &anypb.Any{
+						TypeUrl: httpConnManagerURL,
+						Value: func() []byte {
+							goodHTTPConnManager1 = &httppb.HttpConnectionManager{
+								RouteSpecifier: &httppb.HttpConnectionManager_Rds{
+									Rds: &httppb.Rds{
+										ConfigSource: &basepb.ConfigSource{
+											ConfigSourceSpecifier: &basepb.ConfigSource_Path{
+												Path: "/some/path",
+											},
+										},
+										RouteConfigName: goodRouteName1}}}
+							marshaledConnMgr1, _ = proto.Marshal(goodHTTPConnManager1)
+							return marshaledConnMgr1
```

Comment:
Nit: Prefer renaming `goodHTTPConnManager1` to `cm` and `marshaledConnMgr1` to 'mcm` since the scope of these vars are really small and we could do with small names.

Reply:
Done